modules/magniteAnalyticsAdapter.js

```diff
@@ -749,7 +764,7 @@ const handleBidResponse = (args, bidStatus) => {
 
   // if pbs gave us back a bidId, we need to use it and update our bidId to PBA
   const pbsBidId = (args.pbsBidId == 0 ? generateUUID() : args.pbsBidId) || (args.seatBidId == 0 ? generateUUID() : args.seatBidId);
-  if (pbsBidId) {
+  if (pbsBidId && !cache.bidWasClientSideCached[args.requestId]) {
```

Comment:
Could the entry be cleared after this - to avoid having that table grow indefinitely?

Reply:
for sure was thinking of that - to confirm prebid cannot emit a new bidResponse for same bidId / requestId right?

Was thinking in the scenario a bid adapter returns more than one bidResponse for a bidRequest - need to make sure that this handles that perhaps

Reply:
There Can Be Only One.

I think `args` here is the same object named `bidResponse` in the hook so you could also use a weakset.

Reply:
THERE CAN BE ONLY ONE!!!

Good call - updated to use weakset

thanks!